imports/plugins/core/versions/server/migrations/19_add_object_of_boolean_to_payments_schema.js

```diff
@@ -0,0 +1,120 @@
+import { Migrations } from "meteor/percolate:migrations";
+import { Packages } from "/lib/collections";
+
+const paymentMethodName = {
+  Braintree: "reaction-braintree",
+  AuthNet: "reaction-auth-net",
+  Paypal: "reaction-paypal",
+  Stripe: "reaction-stripe"
+};
+
+const PaypalSupport = {
+  Express: "express",
+  Payflow: "payflow"
+};
+
+Migrations.add({
+  version: 19,
+  up() {
+    // looping through paymentMethodName to get each paymentMethod
+    Object.keys(paymentMethodName).forEach((key) => {
+      let packages;
+      let support;
+      if (paymentMethodName[key] === "reaction-paypal") {
+        packages = Packages.find({ name: paymentMethodName[key] });
+        // Loop through all packages and update support to be an object
+        packages.forEach((pkg) => {
+          // Lopp through all paypalsupport of payment to get each paymentmethod
+          Object.keys(PaypalSupport).forEach((paypal) => {
+            support = pkg.settings[PaypalSupport[paypal]].support;
+            // Checking to make sure support is an array
+            if (Array.isArray(support)) {
+              const supportObject = {};
+              support.forEach((method) => {
+                supportObject[method.toLowerCase()] = true;
```

Comment:
@iakhator You'd also need to deal with the dash in the old value `De-authorize` which will be migrated to `de-authorize`. Expected by schema is `de_authorize`. This results in potential settings losses during migration as seen in screenshot:
![robo_3t_-_1_1_and_men_and__3373_convert_paymentmethod_supports_into_an_object_by_iakhator_ _pull_request__3411_ _reactioncommerce_reaction](https://user-images.githubusercontent.com/1733229/34770263-cce599e2-f600-11e7-9ed0-44909f1073d7.jpg)